### PR TITLE
nextflow: 22.04.5 -> 22.10.4

### LINKED ISSUE #188221: Bash not found when running nextflow project in NixOs
### Describe the bug
Nextflow expects /bin/bash to exists when running projects. This is fixed by 
```ln -s /run/current-system/sw/bin/bash /bin/bash``` as mentionned in nextflow-io/nextflow#1598.

Should it be part of the nix package ?

### Steps To Reproduce
1. Put the following in `main.nf`
```
 process BWA {
     output:
        stdout 
     """
     echo "test"
     """
 }

 workflow {
=    BWA | view
 }
```
2. `nextflow run main.f`

### Expected behavior
A normal execution.

### Additional context
Error message 
```Error executing process > 'BWA'

Caused by:
  Cannot run program "/bin/bash" (in directory "/home/alex/code/bisonex/work/85/f38efc28549cd640297afdeb3f5704"): error=2, No such file or directory

Command executed:

  echo "lol"

Command exit status:
   -
```

### Notify maintainers
@Etjean 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.16.3-microsoft-standard-WSL2, NixOS, 22.11 (Raccoon), 22.11pre401467.8ea014acc33`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(root): `"home-manager, nixos-22.11pre401467.8ea014acc33"`
 - channels(nixos): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

### PR DESCRIPTION
**Pull request by IllustratedMan-code**
(Updates nexflow and adds a FHSUserEnv to fix #188221)

###### Description of changes

Nextflow uses absolute paths to `/bin/bash`. They have decided not to use `/usr/bin/env bash` in some instances, https://github.com/nextflow-io/nextflow/issues/1598. As far as I know, this is impossible to deal with without `buildFHSUserEnv`. I made minimal changes to the existing derivation and added it to a `buildFHSUserEnv`. I also went ahead and updated nextflow to the newest version. There are no significant differences as far as I know.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IllustratedMan-code** [comment]
@ius sorry to call you out on this issue, but I don't seem to be getting anywhere and you helped me in a similar situation before. Is there any way that you could help me get this merged?

File: pkgs/development/interpreters/nextflow/default.nix
Comment (Emiller88) [inline pkgs/development/interpreters/nextflow/default.nix:16]:
```suggestion
    version = "22.10.4";
```
Since it's been a while

**Review by edmundmiller — APPROVED** [review]
Looks good to me. Could we add @IllustratedMan-code and @Emiller88 as maintainers as well to make sure future PRs get reviewed? I'm guessing the overlap of people using nix and nextflow is pretty small 😬

**Comment by IllustratedMan-code** [comment]
@Emiller88 Thanks for reviewing! Do you have the ability to merge?

**Comment by Emiller88** [comment]
I sadly do not have write access :(

**Comment by stianlagstad** [comment]
Thanks for this effort! I could also be added as a maintainer to further secure that future PRs gets reviewed.

**Comment by peterhoeg** [comment]
Saw this on reddit, so let's get it over the finishing line. And I do have write access, so once this is all good we can get it merged.

Couple of things:

1. with the change to .4, the sha256 needs to change
2. the way to properly package nextflow is to fetch the source, patch the various script headers and then build it using our existing java build infrastructure. I suggest you have a look at that after this PR. We have `patchShebangs` for handling script headers in bulk.
3. minor nitpick: with `install -Dm555 $src $out/bin/nextflow` you don't need to create the directory first
4. the previous mkDerivation invocation had some blank lines to make it easier to read, please restore that   
5. if others need to be added to the list of maintainers, please do so
   
Has anyone heard anything from the original maintainer?

**Comment by IllustratedMan-code** [comment]
@peterhoeg I believe I have completed the requested changes. Thanks for the input!

I added the github user names of those who wanted to be maintainers, but don't we need to also add our information to the `maintainer-list.nix` file? Should that be part of this commit?

Also I had to change the java dependency from jre to openjdk17 because jre is now version 19 which nextflow doesn't like. Is there a better way to do this?

**Comment by IllustratedMan-code** [comment]
And, no word from the original maintainer but I'll @Etjean again just in case.

File: pkgs/development/interpreters/nextflow/default.nix
Comment (peterhoeg) [inline pkgs/development/interpreters/nextflow/default.nix:46]: maintainers = with maintainers; [ Etjean IllustratedMan-code Emiller88 stianlagstad ];

**Comment by peterhoeg** [comment]
> I added the github user names of those who wanted to be maintainers, but don't we need to also add our information to the `maintainer-list.nix` file? 

Yes

> Should that be part of this commit?

Separate commit please.

> Also I had to change the java dependency from jre to openjdk17 because jre is now version 19 which nextflow doesn't like. Is there a better way to do this?

That's how you do it.

File: pkgs/development/interpreters/nextflow/default.nix
Comment (IllustratedMan-code) [reply pkgs/development/interpreters/nextflow/default.nix]: Done

**Comment by IllustratedMan-code** [comment]
@peterhoeg I made a separate pull request adding myself as a maintainer here: https://github.com/NixOS/nixpkgs/pull/210967

@Emiller88 If you want to be added to the maintainer list, you have to do the same, otherwise the check will fail.

Thanks again for all the help!

**Comment by peterhoeg** [comment]
Still pending:

1. squash all the commits related to the version bump
2. add a separate commit for adding your id to `maintainer-list.nix`

This PR should have 2 commits in total.

If anyone else is going to be listed as a maintainer, they too need to be in `maintainer-list.nix` as part of commit 2 or they will have to be removed before this will be merged.

**Comment by Emiller88** [comment]
Based on @peterhoeg's comment, I don't think I can push to this branch, and certainly not force push. But please include me in the second commit and add me to `maintainer-list.nix`!

**Comment by K900** [comment]
@IllustratedMan-code looks like you pushed something wrong that caused a ping storm. Please create a new PR to avoid any more mass notifications.

**Pull request closed without merging**
